src/Data/Morpheus/Server/Types/GQLType.hs

```diff
@@ -124,16 +130,17 @@ getTypeConstructorNames = fmap (pack . tyConName . replacePairCon) . getTypeCons
 getTypeConstructors :: Typeable a => f a -> [TyCon]
 getTypeConstructors = ignoreResolver . splitTyConApp . typeRep
 
-deriveTypeData :: Typeable a => f a -> Bool -> TypeCategory -> TypeData
-deriveTypeData proxy shouldPefix cat =
+deriveTypeData :: Typeable a => f a -> (Bool -> String -> String) -> TypeCategory -> TypeData
+deriveTypeData proxy typeNameModifier cat =
   TypeData
-    { gqlTypeName = prefix shouldPefix cat <> getTypename proxy,
+    { gqlTypeName = TypeName . pack $ typeNameModifier (isInput cat) originalTypeName,
```

Comment:
maybe `TypeName . pack $ typeNameModifier (cat == IN) originalTypeName` ?

Reply:
good call